csharp/src/Drivers/BigQuery/BigQueryConnection.cs

```diff
@@ -0,0 +1,954 @@
+
+            return new ParsedDecimalValues()
+            {
+                Precision = Convert.ToInt32(values[0]),
+                Scale = Convert.ToInt32(values[1])
+            };
+        }
+
+        public override IArrowArrayStream GetTableTypes()
+        {
+            StringArray.Builder tableTypesBuilder = new StringArray.Builder();
+            tableTypesBuilder.AppendRange(new string[] { "BASE TABLE", "VIEW" });
+
+            List<IArrowArray> dataArrays = new List<IArrowArray>
+            {
+                tableTypesBuilder.Build()
+            };
+
+            return new BigQueryInfoArrowStream(StandardSchemas.TableTypesSchema, dataArrays, 1);
+        }
+
+        private ListArray CreateNestedListArray(List<IArrowArray> arrayList, IArrowType dataType)
+        {
+            ArrowBuffer.Builder<int> valueOffsetsBufferBuilder = new ArrowBuffer.Builder<int>();
+            ArrowBuffer.BitmapBuilder validityBufferBuilder = new ArrowBuffer.BitmapBuilder();
+            List<ArrayData> arrayDataList = new List<ArrayData>(arrayList.Count);
+            int length = 0;
+            int nullCount = 0;
+
+            foreach (IArrowArray array in arrayList)
+            {
+                if (array == null)
+                {
+                    valueOffsetsBufferBuilder.Append(length);
+                    validityBufferBuilder.Append(false);
+                    nullCount++;
+                }
+                else
+                {
+                    valueOffsetsBufferBuilder.Append(length);
+                    validityBufferBuilder.Append(true);
+                    arrayDataList.Add(array.Data);
+                    length += array.Length;
+                }
+            }
+
+            ArrowBuffer validityBuffer = nullCount > 0
+                ? validityBufferBuilder.Build() : ArrowBuffer.Empty;
+
+            ArrayData data = ArrayDataConcatenator.Concatenate(arrayDataList);
+
+            IArrowArray value = null;
+
+            if (data == null)
+                value = new NullArray(0);
+            else
+                value = ArrowArrayFactory.BuildArray(data);
+
+            valueOffsetsBufferBuilder.Append(length);
+
+            return new ListArray(new ListType(dataType), arrayList.Count,
+                    valueOffsetsBufferBuilder.Build(), value,
+                    validityBuffer, nullCount, 0);
+        }
+
+        public override AdbcStatement CreateStatement()
+        {
+            if (this.client == null || this.credential == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            BigQueryStatement statement = new BigQueryStatement(this.client, this.credential);
+            statement.Options = ValidateOptions();
+            return statement;
+        }
+
+        private IReadOnlyDictionary<string, string> ValidateOptions()
```

Comment:
Should this throw on an unrecognized option?

Reply:
I don't think it should throw anything, but I changed the name to "ParseOptions" instead